### PR TITLE
Don't create lists if in plaintext mode

### LINKED ISSUE #1583: Mail editor creates HTML list even with plain text mode enabled
Enable plain text mode and then write and email with plain text list.

```
* top1
* top2
```

Editor will autocomplete this to a HTML list.

### PR DESCRIPTION
**Pull request by SnowyMouse**
fix #1583

I tested this on both desktop and web and it works. I'm assuming there's nothing more to this?

### CODE REVIEW

File: src/gui/editor/Editor.js [hunk 1]
```diff
@@ -103,17 +104,16 @@ export class Editor implements ImageHandler {
...
-				// Resolve logins.getUserController dependenciy and uncomment block
-				// if (!this._tutanotaProperties.sendPlaintextOnly) {
-				if (e.which === 32) {
-					let blocks = []
-					squire.forEachBlock((block) => {
-						blocks.push(block)
-					})
-					createList(blocks, /^1\.\s$/, true) // create an ordered list if a line is started with '1. '
-					createList(blocks, /^\*\s$/, false) // create an ordered list if a line is started with '1. '
+				if (!logins.getUserController().props.sendPlaintextOnly) {
```
Comment (charlag) [inline src/gui/editor/Editor.js:107]: Depending on globals in UI components is very bad, I think we should pass it in (maybe that's what comment meant, I'm not sure).

**Comment by SnowyMouse** [comment]
I've added a function to Editor that sets whether or not lists are automatically created.

In this case, the only time globals are used is in MailEditor's view function, which it already reads sendPlaintextOnly anyway.

File: src/gui/editor/Editor.js [hunk 2]
```diff
@@ -96,24 +98,28 @@ export class Editor implements ImageHandler {
...
-				// if (!this._tutanotaProperties.sendPlaintextOnly) {
-				if (e.which === 32) {
-					let blocks = []
-					squire.forEachBlock((block) => {
-						blocks.push(block)
-					})
-					createList(blocks, /^1\.\s$/, true) // create an ordered list if a line is started with '1. '
-					createList(blocks, /^\*\s$/, false) // create an ordered list if a line is started with '1. '
+				if (this._createsLists) {
+					if (e.which === 32) {
```
Comment (charlag) [inline src/gui/editor/Editor.js:114]: I forgot to mention that we have Keys constant and `isKeyPressed()` helper.

**Review by charlag — APPROVED** [review]
Yes, this is much better, it's better place to make a decision now.

**Comment by SnowyMouse** [comment]
Hey @charlag. Sorry this one took a while - had lots of stuff to do the past couple of weeks.

Anyway, I used isKeyPressed and replaced the `32` with `Keys.SPACE`. Hope that's better!

I also noticed that the comment for the unordered list was wrong, so I fixed that too.

File: src/gui/editor/Editor.js (hunk 2 above)
Comment (SnowyMouse) [reply src/gui/editor/Editor.js]: I went ahead and took care of those. My latest commit is a64ca9977ac27e403e00f3388206f75eb9f332da now.

**Comment by vitoreiji** [comment]
squashed, rebased and merged in 1982d9f9f86e468e243ade5ec7633b61567c4543

**Pull request closed without merging**